eval/hf_runner.py

```diff
@@ -61,19 +61,36 @@ def get_tokenizer_model(model_name):
             use_cache=True,
             use_flash_attention_2=True,
         )
+    else:
+        print(f"Loading model {model_name}")
+        tokenizer = AutoTokenizer.from_pretrained(model_name)
+        model = AutoModelForCausalLM.from_pretrained(
+            model_name,
+            torch_dtype=torch.float16,
+            trust_remote_code=True,
+            device_map="auto",
+        )
     return tokenizer, model
 
 
-def run_hf_eval(
-    questions_file: str,
-    prompt_file: str,
-    num_questions: int = None,
-    public_data: bool = True,
-    model_name: str = "defog/starcoder-finetune-v3",
-    output_file: str = "results.csv",
-):
+def run_hf_eval(args):
```

Comment:
pass args object wholesale into the function and parse inside it for a tidier interface.